views/js/runner/provider/qti.js

```diff
@@ -190,118 +208,103 @@ define([
 
                 var context = self.getTestContext();
 
-                //to be sure load start after unload...
-                //we add an intermediate ns event on unload
-                self.on('unloaditem.' + action, function(){
-                    self.off('.'+action);
-
-                    self.getProxy()
-                        .callItemAction(context.itemUri, action, params)
-                        .then(function(results){
-
-                            self.setTestContext(results.testContext);
-
-                            if (results.testMap) {
-                                self.setTestMap(results.testMap);
-                            } else {
-                                updateStats();
-                            }
+                //catch server errors
+                var submitError = function submitError(err){
+                    //some server errors are valid, so we don't fail (prevent empty responses)
+                    if(err.code === 200){
+                        self.trigger('alert.submitError',
+                            err.message || __('An error occurred during results submission. Please retry.'),
+                            load
+                        );
+                    } else {
+                        self.trigger('error', err);
+                    }
+                };
 
-                            load();
-                        });
+                //if we have to display modal feedbacks, we submit the responses before the move
+                var feedbackPromise = new Promise( function(resolve){
+                    if(context.hasFeedbacks){
+                        params = _.omit(params, ['itemState', 'itemResponse']);
+
+                        self.getProxy()
+                            .submitItem(context.itemUri, self.itemRunner.getState(), self.itemRunner.getResponses(), params)
+                            .then(function(results){
+                                if (results.itemSession) {
+                                    context.itemAnswered = results.itemSession.itemAnswered;
+
+                                    if(results.displayFeedbacks === true && results.feedbacks) {
+                                        self.itemRunner.renderFeedbacks(results.feedbacks, results.itemSession, function(queue){
+                                            self.trigger('modalFeedbacks', queue, resolve);
+                                        });
+                                        return;
+                                    }
+                                }
+                                return resolve();
+                            });
```

Comment:
There is no `catch` on this promise. What happen if the server refuse the `submit` ?